lib/tasks/test_unit.rake

```diff
@@ -0,0 +1,57 @@
+namespace :test do
```

Comment:
Oh this is interesting! Is this standard from somewhere else?

Reply:
And, just wondering about integration tests and other types of tests?

Reply:
I looked through a lot of resources while doing research on some of the specific errors we were getting in mass and I found that there really is not a way to get them to go away completely, and the solution some developers implement is to turn off the warnings for `Rake::TestTask`. So that part is standard. 

For the names and namespace I used and specific implementation details there is really no standard way to do it. I don't think that `Rake::TestTask` was written with the intention to suppress warnings (there is no variable or so that can be easily passed to suppress warnings) so developers just started creating work arounds that worked for their projects. I came across many different implementations.

I like the solution for a few reasons:
1) it creates the option of running with or without warnings, so regularly warnings can still be checked. Adding on to this because it doesn't override default rake testing options there is a low risk to implementing it. 
2) it will be useful for warnings now and in the future for however long the current implementation of rake testing is still the same. Once it is implemented, if it is done right, it should be pretty low maintenance and auto generate tasks for any new files added. 
3) it's adaptable. **For integration tests**, we could just add that to the functions to include them. `t.pattern` is where you choose the folders. **If we want to add it to plots2**, we will hardly have to tweak anything. Plots2 will provide more insight about performance with using these at a larger scale, too.
4) it's versatile. You have full control over how you run the tests (concurrently or not / which tests / verbose outputs on or off / warnings on or off), which each provide their own useful set of information for debugging.

The easiest way to see if you want to use them and which ones you want is to just run some of them (they all start with the word `autotask` in the description when you run `rake --tasks`)

Reply:
OK, was thinking this over a bit. I like this, but I am pretty cautious about a custom testing configuration. A few things we can do to move forward:

1. let's not do this in the same PR, i think it's just too much and let's finish the mysql work here, if that's OK?
2. let's add integration tests for sure
3. can you, in a comment above each one, explain the command to run each task?
4. i do think we will need to get this running in Travis if we want to merge it, as that will show people that it's working as intended.

Thanks, @sashadev-sky! I hope you don't mind my caution on developing customizations, i am just thinking of myself reading this or puzzling through it 5+ years in the future 😂  and want to make a considered decision!

Reply:
@jywarren Ok got it. I will take it out and address your comments in a new PR. No problem! It's better to be cautious. 

I'll update my PR description to reflect only small changes I made to get rid of small errors, and add the notes about warning reductions to the new PR.

For the mysql updates, I can instruct people what version of mysql to use but I don't have any information on the production database. Does it use mariadb? what version?  Where would I figure out this information from myself? Mysql and mariadb are pretty similar but it's good to check version compatibility between them too.